app/models/conversation_user.rb

```diff
@@ -20,4 +22,28 @@
 class ConversationUser < ApplicationRecord
   belongs_to :user
   belongs_to :conversation
+
+  validates :unread_messages, numericality: { only_integer: true }
+
+  def self.conversations_unread_msgs(user_id)
+    array = ConversationUser.where('user_id = ?', user_id)
+    Hash[array.map do |conv_user|
+      [conv_user.conversation_id, conv_user.unread_messages]
+    end]
+  end
+
+  def self.new_message(conversation_id, user_id)
+    instances = ConversationUser.where('conversation_id = ?', conversation_id)
+                                .where('user_id != ?', user_id)
+    instances.each do |instance|
+      instance.update!(unread_messages:
```

Comment:
Do you want me to move 
`new_message`, and `read_messages` logic to the conversation model? 
If that is the case, I cant agree, since both functions are modifying ConversationUser instances, instead of Conversation ones. 
@leandrodzp cant tag 😞

Reply:
No, I'm talking about
```ruby 
conversation_user.update!(unread_messages:
          (conversation_user.unread_messages + 1))
```
You can make a method like `incremet_unread(number)`, WDYT?

Reply:
As we talked via Slack, let's keep this this way since it makes sense with the way you implemented all the other stuff.